src/once/index.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Unit,
+  Store,
+  Event,
+  Effect,
+  sample,
+  createStore,
+  EventAsReturnType,
+} from 'effector';
+
+export function once<T>(
+  unit: Event<T> | Effect<T, any, any> | Store<T>,
+): EventAsReturnType<T> {
+  const $canTrigger = createStore<boolean>(true);
+
+  const trigger: Event<T> = sample({
+    clock: unit as Unit<T>,
+    filter: $canTrigger,
+  });
+
+  $canTrigger.on(trigger, () => false);
+
+  return sample({ clock: trigger });
```

Comment:
Why do you need second `sample`?

Reply:
This does not allow `target` to influence `$canTrigger` if called directly. This will be redundant when `DerivedEvent` arrives, I suppose.

Reply:
@sergeysova Do you think it is better to remove it? I'm okay with it, especially because calling `trigger` emits a warning